compiler/src/lib.rs

```diff
@@ -223,6 +223,8 @@ pub fn inputs_to_query_callback<T: FieldElement>(inputs: Vec<T>) -> impl Fn(&str
                 let value = inputs.get(index).cloned();
                 if let Some(value) = value {
                     log::trace!("Input query: Index {index} -> {value}");
+                } else {
+                    panic!("Not enough inputs provided! Index {index} out of bounds")
```

Comment:
Are there situations where this is a recoverable error? I guess this could be a warning as well.

Reply:
Please don't panic here. It is recoverable depending on the pil.

Reply:
We need to improve the prover input in general, it should be way more complex.

Reply:
OK, will make it warning!